dashboard/app/handler.go

```diff
@@ -68,10 +73,27 @@ func handleContext(fn contextHandler) http.Handler {
 }
 
 type (
-	ErrDontLog  struct{ error }
+	ErrClient   struct{ error }
 	ErrRedirect struct{ error }
 )
 
+var ErrClientNotFound = &ErrClient{errors.New("resource not found")}
+var ErrClientBadRequest = &ErrClient{errors.New("bad request")}
+
+func (ce *ErrClient) HTTPStatus() int {
+	switch ce {
+	case ErrClientNotFound:
+		return http.StatusNotFound
+	case ErrClientBadRequest:
+		return http.StatusBadRequest
+	}
+	return http.StatusInternalServerError
+}
+
+func (ce *ErrClient) WithDetails(err error) error {
```

Comment:
I think it somewhat makes it look like the actual error wrapping is reversed. In reality we wrap e.g. `ErrClientNotFound` error into some other error, then the resulting error into something else, etc. But `WithDetails` seems as if we add something into the `ErrClientNotFound` error itself, which is not true.

I think it would be cleaner if there were no `WithDetails` function and instead you just invoked fmt.Errorf() whenever you needed it.

Reply:
fmt.Errorf() wrapping ErrClient then. Ok.